### DISCUSSION TITLE
Support for Core200S "night light?"

### DISCUSSION THREAD

**Discussion by khanbarfuss** [RaresAil/homebridge-levoit-air-purifier#34] [Ideas]
Hey there, using for both 400S and 200. Work great! The 200S has a light on top that can be controlled via the VeSync app, I was wondering if it would be possible to expose this within HomeKit. Could be super handy to setup an automation with a "night light" using the air purifier. Thanks!

**Comment by RaresAil** [comment]
Hey, the last speed level in the app should be the night mode, can you try that?

**Reply by khanbarfuss** [reply]
Hey thanks for getting back. 
The speed levels within HomeKit go 0-off, 1-"Sleep Mode" , 2-Low, 3-Medium, 4-High. Which correspond to VeSync app with fan speed. 
But in VeSync app below the Display toggle, there is a "Night Light" option, which has "off" "50%" and "100%" as available choices. But with the plugin using HomeKit, there is not a switch that triggers the light by itself. (within app this night light can be turned on even if the fan is off- shown in pic below)

I'm curious if this night light can be exposed as an actual lightbulb within HomeKit so that it can work as a trigger and be able to be part of automations without the fan needing to be turned on.

![IMG_4320](https://user-images.githubusercontent.com/102782094/173945681-3f6bd7dc-7cea-49b0-bd88-3c80f5be3482.jpg)

**Reply by RaresAil** [reply]
Hello, can you [enable the debug mode](https://github.com/RaresAil/homebridge-levoit-air-purifier#enabling-debug-mode)? and send some logs, and remove any private information from there if there are any.

**Reply by RaresAil** [reply]
It looks like the 200s has the night light manually, the other versions is automatically on the sleep mode

**Reply by khanbarfuss** [reply]
Well the other ones don't have a specific "night light" inside of them. This is different from the night mode setting which just turns the display dark and the speed low. This is a standalone light built in to the 200S - so the rest of the air purifier (fan) can be turned off, but a light that surrounds the top can be enabled to give a night light for the room. 

Here's some logs with MAC addresses and UUIDs removed. 

`[22/07/2022, 13:02:22] [Levoit Air Purifiers] [DEBUG]: [GET DEVICES] Device List -> JSON: [{"deviceRegion":"US","isOwner":true,"authKey":null,"deviceName":"Bedroom Air Ourifier","deviceImg":"https://image.vesync.com/defaultImages/Core_200S/icon_200s_purifier_160.png","cid":"vsaqb5da73e611eb876500163e0cf75c","deviceStatus":"on","connectionStatus":"online","connectionType":"wifi+BTOnboarding","deviceType":"Core200S","type":"wifi-air","uuid":"UUIDREMOVED","configModule":"WiFiBTOnboarding_AirPurifier_Core200S_US","macID":"MACADDRESSREMOVED","mode":null,"speed":null,"extension":{"airQuality":-1,"airQualityLevel":1,"mode":"manual","fanSpeedLevel":"1"},"currentFirmVersion":null,"subDeviceNo":null,"subDeviceType":null,"deviceFirstSetupTime":"Jun 9, 2022 12:27:39 AM","deviceProp":null},{"deviceRegion":"US","isOwner":true,"authKey":null,"deviceName":"Living Room","deviceImg":"https://image.vesync.com/defaultImages/Core_400S_Series/icon_core400s_purifier_160.png","cid":"vsaqd3c67c09432fa4f0a0493f74583c","deviceStatus":"on","connectionStatus":"online","connectionType":"WiFi+BTOnboarding+BTNotify","deviceType":"LAP-C401S-WUSR","type":"wifi-air","uuid":"UUIDREMOVED","configModule":"WiFiBTOnboardingNotify_AirPurifier_LAP-C401S-WUSR_US","macID":"MACADDRESSREMOVED","mode":null,"speed":null,"extension":{"airQuality":-1,"airQualityLevel":1,"mode":"auto","fanSpeedLevel":"1"},"currentFirmVersion":null,"subDeviceNo":null,"subDeviceType":null,"deviceFirstSetupTime":"Apr 1, 2022 11:25:12 PM","deviceProp":null}]`

**Reply by RaresAil** [reply]
I don't find any info about the light feature in those logs, do you have more logs? cand try to get the logs with the light open

**Reply by khanbarfuss** [reply]
When toggling the light on and off no logs come through with the debug flag enabled. With the light on and the Homebridge service restarted I gathered the following, but I don't see any indication of a light in these either.

`[26/07/2022, 10:37:46] [Levoit Air Purifiers] [DEBUG]: [GET DEVICES] Device List -> JSON: [{"deviceRegion":"US","isOwner":true,"authKey":null,"deviceName":"Bedroom Air Purifier","deviceImg":"https://image.vesync.com/defaultImages/Core_200S/icon_200s_purifier_160.png","cid":"vsaqb5da73e611eb876500163e0cf75c","deviceStatus":"on","connectionStatus":"online","connectionType":"wifi+BTOnboarding","deviceType":"Core200S","type":"wifi-air","uuid":"UUIDREMOVED","configModule":"WiFiBTOnboarding_AirPurifier_Core200S_US","macID":"MACADDRESSREMOVED","mode":null,"speed":null,"extension":{"airQuality":-1,"airQualityLevel":1,"mode":"manual","fanSpeedLevel":"1"},"currentFirmVersion":null,"subDeviceNo":null,"subDeviceType":null,"deviceFirstSetupTime":"Jun 9, 2022 12:27:39 AM","deviceProp":null},{"deviceRegion":"US","isOwner":true,"authKey":null,"deviceName":"Living Room","deviceImg":"https://image.vesync.com/defaultImages/Core_400S_Series/icon_core400s_purifier_160.png","cid":"vsaqd3c67c09432fa4f0a0493f74583c","deviceStatus":"on","connectionStatus":"online","connectionType":"WiFi+BTOnboarding+BTNotify","deviceType":"LAP-C401S-WUSR","type":"wifi-air","uuid":"UUIDREMOVED","configModule":"WiFiBTOnboardingNotify_AirPurifier_LAP-C401S-WUSR_US","macID":"MACADDRESSREMOVED","mode":null,"speed":null,"extension":{"airQuality":-1,"airQualityLevel":1,"mode":"manual","fanSpeedLevel":"1"},"currentFirmVersion":null,"subDeviceNo":null,"subDeviceType":null,"deviceFirstSetupTime":"Apr 1, 2022 11:25:12 PM","deviceProp":null}]`

**Reply by RaresAil** [reply]
If there is no info about a night light, i can't get the status of it

**Reply by Sir-Ian** [reply]
I have the same model also wondering about the nightlight. 

My home app only controls "air cleaner display", what logs do you need?

**Comment by RaresAil** [comment]
Hey @khanbarfuss i made a discovery on this, there is a possibility to add night light feature, can you provide some more logs using the version 1.2.3 ?

**Reply by almonde** [reply]
Hi RaresAli, here's a debug from 1.30:

Light off:
`[4/27/2023, 4:51:55 PM] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] JSON: {"traceId":"1682632314948","code":0,"msg":"request success","module":null,"stacktrace":null,"result":{"traceId":"1682632314948","code":0,"result":{"enabled":true,"filter_life":91,"mode":"manual","level":1,"display":true,"child_lock":false,"night_light":"off","configuration":{"display":true,"display_forever":false},"device_error_code":0}}} `

And then with the light switch on at 100%:
`[4/27/2023, 4:54:14 PM] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] JSON: {"traceId":"1682632453819","code":0,"msg":"request success","module":null,"stacktrace":null,"result":{"traceId":"1682632453819","code":0,"result":{"enabled":true,"filter_life":91,"mode":"manual","level":1,"display":true,"child_lock":false,"night_light":"on","configuration":{"display":true,"display_forever":false},"device_error_code":0}}} `

Light switched on in low/dim setting:
`[4/27/2023, 4:56:31 PM] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] JSON:
 {"traceId":"1682632590981","code":0,"msg":"request success","module":null,"stacktrace":null,"result":{"traceId":"1682632590981","code":0,"result":{"enabled":true,"filter_life":91,"mode":"manual","level":1,"display":true,"child_lock":false,"night_light":"dim","configuration":{"display":true,"display_forever":false},"device_error_code":0}}} `

So at least its status is showing now.

**Reply by khanbarfuss** [reply]
Hey @RaresAil sorry was out for a bit, but here are some logs that coincide with @almonde's findings with v1.3.0

Off:

```
[03/05/2023, 10:41:47] [Levoit Air Purifiers] [DEBUG]: [LOGIN] Logging in...
[03/05/2023, 10:41:47] [Levoit Air Purifiers] [DEBUG]: [LOGIN] The authentication success
[03/05/2023, 11:11:11] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] Getting device info...
[03/05/2023, 11:11:12] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] JSON: {"traceId":"1683133871362","code":0,"msg":"request success","module":null,"stacktrace":null,"result":{"traceId":"1683133871362","code":0,"result":{"enabled":false,"filter_life":55,"mode":"manual","level":1,"display":false,"child_lock":false,"night_light":"off","configuration":{"display":true,"display_forever":false},"device_error_code":0}}}
```

Dim setting:
```
[03/05/2023, 13:57:39] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] Getting device info...
[03/05/2023, 13:57:40] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] JSON: {"traceId":"1683143859551","code":0,"msg":"request success","module":null,"stacktrace":null,"result":{"traceId":"1683143859551","code":0,"result":{"enabled":false,"filter_life":55,"mode":"manual","level":1,"display":false,"child_lock":false,"night_light":"dim","configuration":{"display":true,"display_forever":false},"device_error_code":0}}}
```

On/High setting:

```
[03/05/2023, 13:59:16] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] Getting device info...
[03/05/2023, 13:59:17] [Levoit Air Purifiers] [DEBUG]: [GET DEVICE INFO] JSON: {"traceId":"1683143956938","code":0,"msg":"request success","module":null,"stacktrace":null,"result":{"traceId":"1683143956938","code":0,"result":{"enabled":false,"filter_life":55,"mode":"manual","level":1,"display":false,"child_lock":false,"night_light":"on","configuration":{"display":true,"display_forever":false},"device_error_code":0}}}
```

**Reply by RaresAil** [reply]
Hey, thank you for the logs, can one of you create an issue for this with the logs? To be easier to track

**Reply by khanbarfuss** [reply]
You bet!
Done ✅  "Feature Request: Core200S Night Light #76"
https://github.com/RaresAil/homebridge-levoit-air-purifier/issues/76
